### DISCUSSION TITLE
Best way to access the API?

### DISCUSSION THREAD

**Discussion by odysseus654** [richardthombs/scunpacked#58] [Q&A]
It feels like the information in here has the potential for a lot of useful stuff.  I have some ideas that I'd like to explore and I'm unsure as to the best way to handle the APIs in here.  I'm assuming that the APIs are in the end static files (and more likely just commits to this github repo and published to the scunpacked.com website).  I'm also assuming that the files will generally be kept up to date with each released patch on the site.

I'm trying to roll through my head how the best way to handle this might be.  Would you have objections to my using the scunpacked.com website as an [origin](https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_Origin.html) for a [CloudFront](https://en.wikipedia.org/wiki/Amazon_CloudFront) distribution containing my javascript-only app for instance?

**Comment by richardthombs** [comment]
Hi there,

Yep the API is currently just a bunch of static json files. Feel free to use them downstream however you like. Some accreditation would boost my ego and obviously all the data really belongs to CIG, so please follow their community project guidelines as well.

I try to keep it up to date with every patch, starting with wave 1 PTU.

If you are going to build something from the API now, please use the v2 folder, because the earlier version will be going away once I've fully transitioned the website.

**Reply by odysseus654** [reply]
Okay, think I pulled the repo like a day after you added the v2 stuff and was wondering if that fully replaced v1, guess I have my answer. Also assuming that the generator code is the best place to get documentation of what all that info is (when there is a meaning outside of "it came from CIG").

I'm gonna have to think of a new domain name for this, ugh.

A little curious about all the "connections" in the files, wondering if they eventually will add the connection nodes to this that they've been discussing for a little while.

**Reply by richardthombs** [reply]
The API is still evolving quite frequently, so I don't really want to spend time documenting it until it is a bit more stable. Once you put a bit of time into it, I think it will become quite clear.

I've not finished v2 yet, the main thing missing are the item indexes like items.json, ship-items.json etc.

What connections are you referring to?

**Reply by odysseus654** [reply]
Yah, I can understand that it's evolving quickly (as it likely should be at this point).

FWIW I have created a CloudFront mirror of scunpacked.com/api under scanalysis.app/api (using heavy caching to reduce hits to your server) although I'm not using it for anything yet.  I'm not entirely sure what I'm doing with this yet, it's mostly a combination of me wondering where my EM/IR reactions are coming from (especially when they don't get described at ERKUL) as well as seeing the contents of the .json files you have which could show how the flow of heat comes from each component (even the ones we can't change yet, like the engines)

The connections I was referring to appear only in the "raw" files and are called "pipes".  I'm assuming these are coming directly from CIG and I'm curious as to whether they would eventually change over into the Nodes mentioned as upcoming ![Nodes Diagram](https://i.redd.it/14q9a12zo3w51.png)

**Comment by richardthombs** [comment]
Ah yes, that awesome demo. I don’t think any of that is in the game data yet. The closest so far are the “pipes”, but they seem to be more of a common “bus” where everything gets plugged into the heat / power / fuel pipe and then can draw what it wants. There doesn’t seem to be any kind of graph between the nodes yet. R.--Richard Thombsgearstone.uk From: Heather AndersonSent: 21 December 2020 10:03To: richardthombs/scunpackedCc: Richard Thombs; CommentSubject: Re: [richardthombs/scunpacked] Best way to access the API? (#58) Yah, I can understand that it's evolving quickly (as it likely should be at this point).FWIW I have created a mirror of scunpacked.com/api under scanalysis.app/api (although I'm not using it for anything yet). I'm not entirely sure what I'm doing yet, it's mostly a combination of me wondering where my EM/IR reactions are coming from (especially when they don't get described at ERKUL) as well as seeing the contents of the .json files you have which could show how the flow of heat comes from each component (even the ones we can't change yet, like the engines)The connections I was referring to appear only in the "raw" files and are called "pipes". I'm assuming these are coming directly from CIG and I'm curious as to whether they would eventually change over into the Nodes mentioned as upcoming —You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.
